skate-plugin/src/main/kotlin/com/slack/sgp/intellij/circuitgenerator/UdfViewModelGenerator.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2024 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.slack.sgp.intellij.circuitgenerator
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.ui.components.dialog
+import com.intellij.ui.dsl.builder.bindSelected
+import com.intellij.ui.dsl.builder.bindText
+import com.intellij.ui.dsl.builder.panel
+import com.slack.sgp.intellij.util.getJavaPackageName
+import com.slack.sgp.intellij.util.isCircuitGeneratorEnabled
+import java.io.File
+import slack.tooling.projectgen.circuitgen.CircuitPresenter
+import slack.tooling.projectgen.circuitgen.CircuitScreen
+
+class UdfViewModelGenerator : AnAction(), DumbAware {
```

Comment:
Are we adding component generation for UdfViewModel in this PR too?

Reply:
Oops, yup i'm adding UdfViewModel as well. Updated in https://github.com/slackhq/slack-gradle-plugin/pull/799/commits/6986e693d5d0bf42d69c263050612fb0634dbdbc

Reply:
Sorry I meant this as sort of a rhetorical question, I don't think we should be 😅